be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -840,6 +841,14 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
     auto fragment_ctx = context->fragment_context();
     if (typeid(*datasink) == typeid(starrocks::ResultSink)) {
         auto* result_sink = down_cast<starrocks::ResultSink*>(datasink.get());
+
+        // Accumulate chunks before sending to result sink
+        if (runtime_state->query_options().__isset.enable_result_sink_accumulate &&
+            runtime_state->query_options().enable_result_sink_accumulate) {
+            fragment_ctx->pipelines().back()->add_op_factory(
+                    std::make_shared<ChunkAccumulateOperatorFactory>(context->next_operator_id(), 0));
```

Comment:
not sure if 0 works since result sink does not have plan node id.

The second argument is supposed to be plan node id, but result sink does not have plan node id.

```
class ChunkAccumulateOperatorFactory final : public OperatorFactory {
public:
    ChunkAccumulateOperatorFactory(int32_t id, int32_t plan_node_id)
            : OperatorFactory(id, "chunk_accumulate", plan_node_id) {}

    ~ChunkAccumulateOperatorFactory() override = default;

    OperatorPtr create(int32_t degree_of_parallelism, int32_t driver_sequence) override {
        return std::make_shared<ChunkAccumulateOperator>(this, _id, _plan_node_id, driver_sequence);
    }
};

```

Plus, you can also conside to use

`ExecNode::may_add_chunk_accumulate_operator`

Reply:
Good catch! I will use ` Operator::s_pseudo_plan_node_id_for_final_sink` as plan node id instead.